etcd3locker/lock.go

```diff
@@ -0,0 +1,47 @@
+// Tested on etcd 3.1+
+package etcd3locker
+
+import (
+	"context"
+	"time"
+
+	"github.com/coreos/etcd/clientv3/concurrency"
+	"github.com/tus/tusd"
+)
+
+type Etcd3Lock struct {
```

Comment:
Would it be possible to not export this type since it's just an implementation detail?

Reply:
Yes - I've made this change in a later commit to not export this type.